.github/workflows/unit_test_cpu.yaml

```
@@ -0,0 +1,46 @@
+      max-parallel: 20
+      matrix:
+        os: [ubuntu-22.04]
+        python-version: [3.10.16]
+    steps:
+      - name: Cache Python Packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ matrix.python-version }}-${{ hashFiles('**/pyproject.toml') }}
+      - name: Setup Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)